x-pack/test/apm_api_integration/tests/alerts/transaction_error_rate.spec.ts

```diff
@@ -82,38 +66,19 @@ export default function ApiTest({ getService }: FtrProviderContext) {
 
     after(async () => {
       await synthtraceEsClient.clean();
-      await supertest.delete(`/api/alerting/rule/${ruleId1}`).set('kbn-xsrf', 'foo');
-      await supertest.delete(`/api/actions/connector/${actionId1}`).set('kbn-xsrf', 'foo');
-      await supertest.delete(`/api/alerting/rule/${ruleId2}`).set('kbn-xsrf', 'foo');
-      await supertest.delete(`/api/actions/connector/${actionId2}`).set('kbn-xsrf', 'foo');
-      await esDeleteAllIndices([ALERT_ACTION_INDEX_NAME1, ALERT_ACTION_INDEX_NAME2]);
-      await es.deleteByQuery({
-        index: APM_ALERTS_INDEX,
-        query: { term: { 'kibana.alert.rule.uuid': ruleId1 } },
-      });
-      await es.deleteByQuery({
-        index: APM_ALERTS_INDEX,
-        query: { term: { 'kibana.alert.rule.uuid': ruleId2 } },
-      });
-      await es.deleteByQuery({
-        index: '.kibana-event-log-*',
-        query: { term: { 'kibana.alert.rule.consumer': 'apm' } },
-      });
+      await clearKibanaApmEventLog(es);
     });
 
     describe('create alert without filter query', () => {
+      let ruleId: string;
+
       before(async () => {
-        actionId1 = await createIndexConnector({
-          supertest,
-          name: 'Transation error rate without filter query',
-          indexName: ALERT_ACTION_INDEX_NAME1,
-        });
```

Comment:
I've removed all reference to the index connector action. We don't need it to test whether an alert was produced since we can just query the alerts index directly

Reply:
There are two things that we are testing: 
- whether an alert is triggered (and shown in Kibana UI).
- alert notification was sent (with index connector in this case) with correct action variables.

I think the tests to check alert notifications are useful.

Reply:
> alert notification was sent (with index connector in this case) with correct action variables.

Are we testing the APM rule in this case, or the connector functionality (and thus the alerting framework). It sounds like we are testing the latter where we should focus on the former.

Reply:
.. or put another way: what is currently not covered anymore? What could break _in the APM rule_ that would no longer go noticed?

Reply:
We are setting `context` action variables during rule execution https://github.com/elastic/kibana/blob/f7e36a96d035e1cb4feec334a3bab270be9eac4e/x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts#L288C69-L288C69.

In the tests, we are checking if the action variables are set correctly in alert notification. Without it, any changes around `context` action variables would not be checked anymore.